packages/alfa-dom/src/tables/cell.ts

```diff
@@ -0,0 +1,703 @@
+import { Equatable } from "@siteimprove/alfa-equatable";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { Serializable } from "@siteimprove/alfa-json";
+import { Map } from "@siteimprove/alfa-map";
+import { None, Option, Some } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+
+import * as json from "@siteimprove/alfa-json";
+import { Err, Ok, Result } from "@siteimprove/alfa-result";
+
+import { BuildingTable, Document, Element } from "..";
+import {
+  hasName,
+  isDescendantOf,
+  isElementByName,
+  isEmpty,
+  isEqual,
+  parseAttribute,
+  parseEnumeratedAttribute,
+  parseSpan,
+  parseTokensList,
+  resolveReferences,
+} from "./helpers";
+
+const { some } = Iterable;
+const { and, equals, not } = Predicate;
+
+/**
+ * @see https://html.spec.whatwg.org/multipage/tables.html#concept-cell
+ */
+export class Cell implements Equatable, Serializable {
+  private readonly _kind: Cell.Kind;
+  private readonly _anchorX: number;
+  private readonly _anchorY: number;
+  private readonly _width: number;
+  private readonly _height: number;
+  private readonly _element: Element;
+  private readonly _headers: Array<Element>;
+
+  public static of(
+    kind: Cell.Kind,
+    x: number,
+    y: number,
+    w: number,
+    h: number,
+    element: Element,
+    headers: Array<Element> = []
+  ): Cell {
+    return new Cell(kind, x, y, w, h, element, headers);
+  }
+
+  private constructor(
+    kind: Cell.Kind,
+    x: number,
+    y: number,
+    w: number,
+    h: number,
+    element: Element,
+    headers: Array<Element>
+  ) {
+    this._kind = kind;
+    this._anchorX = x;
+    this._anchorY = y;
+    this._width = w;
+    this._height = h;
+    this._element = element;
+    this._headers = headers;
+  }
+
+  // debug
+  public get name(): string {
+    return this._element.attribute("id").get().value;
+  }
+
+  public get anchor(): { x: number; y: number } {
+    return { x: this._anchorX, y: this._anchorY };
+  }
+  public get width(): number {
+    return this._width;
+  }
+  public get height(): number {
+    return this._height;
+  }
+  public get kind(): Cell.Kind {
+    return this._kind;
+  }
+  public get element(): Element {
+    return this._element;
+  }
+  public get headers(): Iterable<Element> {
+    return this._headers;
+  }
+
+  public isCovering(x: number, y: number): boolean {
+    return !(
+      // cell is *not* covering if either
+      (
+        x < this._anchorX || // slot is left of cell
+        this._anchorX + this._width - 1 < x || // slot is right of cell
+        y < this._anchorY || // slot is above cell
+        this._anchorY + this._height - 1 < y
+      ) // slot is below cell
+    );
+  }
+
+  /**
+   * compare cell according to their anchor
+   * in a given group of cells (row, rowgroup, table, …), no two different cells can have the same anchor, so this is good.
+   */
+  public compare(cell: Cell): number {
```

Comment:
I've added a `Comparable<T>` interface that this class should implement: https://github.com/Siteimprove/alfa/blob/master/packages/alfa-comparable/src/comparable.ts.